libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c

```diff
@@ -148,459 +116,264 @@ static bool prvAreUpdateIdsEqual(
+            &xBaseUpdateManifest );
 
-        if( ( pxAduClient->pxLastInstallResult->ucResultDetails != NULL ) && ( pxAduClient->pxLastInstallResult->ulResultDetailsLength > 0 ) )
-        {
-            xLastInstallResult.result_details = az_span_create( ( uint8_t * ) pxAduClient->pxLastInstallResult->ucResultDetails, pxAduClient->pxLastInstallResult->ulResultDetailsLength );
-        }
-        else
+        if( az_result_failed( xAzResult ) )
         {
-            xLastInstallResult.result_details = AZ_SPAN_EMPTY;
+            AZLogError( ( "az_iot_adu_ota_parse_update_manifest failed: 0x%08x", xAzResult ) );
+            /* TODO: return individualized/specific errors. */
+            return eAzureIoTErrorFailed;
         }
 
-        xLastInstallResult.step_results_count = pxAduClient->pxLastInstallResult->ulStepResultsCount;
-
-        for( int lIndex = 0; lIndex < pxAduClient->pxLastInstallResult->ulStepResultsCount; lIndex++ )
-        {
-            xLastInstallResult.step_results[ lIndex ].step_id = az_span_create(
-                ( uint8_t * ) pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ucStepId,
-                pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ulStepIdLength );
-
-            xLastInstallResult.step_results[ lIndex ].result_code =
-                pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ulResultCode;
-            xLastInstallResult.step_results[ lIndex ].extended_result_code =
-                pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ulExtendedResultCode;
-
-            if( ( pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ucResultDetails != NULL ) &&
-                ( pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ulResultDetailsLength > 0 ) )
-            {
-                xLastInstallResult.step_results[ lIndex ].result_details = az_span_create(
-                    ( uint8_t * ) pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ucResultDetails,
-                    pxAduClient->pxLastInstallResult->pxStepResults[ lIndex ].ulResultDetailsLength
-                    );
-            }
-            else
-            {
-                xLastInstallResult.step_results[ lIndex ].result_details = AZ_SPAN_EMPTY;
-            }
-        }
+        prvCastUpdateRequest( &xBaseUpdateRequest, &xBaseUpdateManifest, pxAduUpdateRequest );
     }
 
-    if( az_result_failed(
-            az_iot_adu_ota_get_properties_payload(
-                &pxAduClient->pxHubClient->_internal.xAzureIoTHubClientCore,
-                &xDeviceInformation,
-                ( int32_t ) pxAduClient->xState,
-                &pxAduClient->xUpdateRequest.workflow,
-                ( pxAduClient->pxLastInstallResult != NULL ) ? &xLastInstallResult : NULL,
-                xPropertiesPayload,
-                &xPropertiesPayload ) ) )
+    return eAzureIoTSuccess;
+}
+
+AzureIoTResult_t AzureIoTADUClient_GetResponse( AzureIoTHubClient_t * pxAzureIoTHubClient,
```

Comment:
was this the API which we had decided to be a `SendResponse()`?

Reply:
Yes. It's now changed to the SendResponse pattern we decided for.